ttnn/cpp/ttnn/operations/experimental/transformer/rotary_embedding_llama_fused_qk/device/rotary_embedding_llama_fused_qk_program_factory.cpp

```diff
@@ -0,0 +1,251 @@
+    uint32_t num_trans_mat_tiles = 1;
+    tt_metal::CircularBufferConfig cb_trans_mat_config =
+        tt_metal::CircularBufferConfig(num_trans_mat_tiles * trans_mat_single_tile_size, {{trans_mat_cb_index, trans_mat_cb_data_format}})
+            .set_page_size(trans_mat_cb_index, trans_mat_single_tile_size).
+            set_globally_allocated_address(*trans_mat_buffer);
+    auto cb_trans_mat = tt_metal::CreateCircularBuffer(program, all_cores, cb_trans_mat_config);
+
+    uint32_t num_interm_tiles = head_dim_t;
+    uint32_t rotated_input_interm_cb_index = CB::c_intermed0;
+    tt_metal::CircularBufferConfig cb_rotated_input_interm_config =
+        tt_metal::CircularBufferConfig(
+            num_interm_tiles * input_single_tile_size, {{rotated_input_interm_cb_index, input_cb_data_format}})
+            .set_page_size(rotated_input_interm_cb_index, input_single_tile_size);
+    auto cb_rotated_input_interm = tt_metal::CreateCircularBuffer(program, all_cores, cb_rotated_input_interm_config);
+
+    uint32_t cos_interm_cb_index = CB::c_intermed1;
+    tt_metal::CircularBufferConfig cb_cos_interm_config =
+        tt_metal::CircularBufferConfig(
+            num_interm_tiles * input_single_tile_size, {{cos_interm_cb_index, cos_cb_data_format}})
+            .set_page_size(cos_interm_cb_index, cos_single_tile_size);
+    auto cb_cos_interm = tt_metal::CreateCircularBuffer(program, all_cores, cb_cos_interm_config);
+
+    uint32_t sin_interm_cb_index = CB::c_intermed2;
+    tt_metal::CircularBufferConfig cb_sin_interm_config =
+        tt_metal::CircularBufferConfig(
+            num_interm_tiles * input_single_tile_size, {{sin_interm_cb_index, sin_cb_data_format}})
+            .set_page_size(sin_interm_cb_index, sin_single_tile_size);
+    auto cb_sin_interm = tt_metal::CreateCircularBuffer(program, all_cores, cb_sin_interm_config);
+
+    uint32_t q_output_cb_index = CB::c_out0;  // output operands start at index 16
+    tt_metal::CircularBufferConfig cb_q_output_config =
+        tt_metal::CircularBufferConfig(
+            num_q_output_tiles * output_single_tile_size, {{q_output_cb_index, output_cb_data_format}})
+            .set_page_size(q_output_cb_index, output_single_tile_size)
+            .set_globally_allocated_address(*q_dst_buffer);
+    auto cb_q_output = tt_metal::CreateCircularBuffer(program, q_cores, cb_q_output_config);
+    uint32_t k_output_cb_index = CB::c_out1;  // output operands start at index 17
+    tt_metal::CircularBufferConfig cb_k_output_config =
+        tt_metal::CircularBufferConfig(
+            num_k_output_tiles * output_single_tile_size, {{k_output_cb_index, output_cb_data_format}})
+            .set_page_size(k_output_cb_index, output_single_tile_size)
+            .set_globally_allocated_address(*k_dst_buffer);
+    auto cb_k_output = tt_metal::CreateCircularBuffer(program, k_cores, cb_k_output_config);
+
+
+    // Set up the kernel
+    std::vector<uint32_t> q_compute_kernel_args = {
```

Comment:
We don't create vectors for kernel args anymore, use an array instead. And you don't need the explicit C-style cast for every value.

Reply:
I see an error, when I try to use an array. Isn't compile_args in create kernel() needs vector<uint32_t> https://github.com/tenstorrent/tt-metal/blob/5f1589291d4a2f074fbf098070edbe3fa1f8734a/tests/tt_metal/tt_metal/unit_tests_fast_dispatch/common/command_queue_fixture.hpp#L388

Reply:
No, you just need to remove the `<uint32_t>` template argument to make CTAD work, this is the most likely issue.